File: Georechner_C++/grundlagen/winkel.cpp
#include <math.h>
#include "mainwindow.h"

float rho_deg = 180.0f / M_PI, rho_gon = 200.0f / M_PI;

float rad2gon(float& p_rad) {return p_rad * rho_gon;}
long double rad2gon(long double& p_rad) {return p_rad * rho_gon;}

float rad2deg(float& p_rad) {return p_rad * rho_deg;}

float gon2rad(float& p_gon) {return p_gon / rho_gon;}

float gon2deg(float& p_gon) {float winkel_rad = gon2rad(p_gon); return rad2deg(winkel_rad);}

float deg2rad(float& p_deg) {return p_deg / rho_deg;}

float deg2gon(float& p_deg) {float winkel_deg = deg2rad(p_deg); return rad2gon(winkel_deg);}

float cot(float& p_ctg) {return cos(p_ctg)/sin(p_ctg);}

float richtungswinkel_aus_richtung(float t, float r) {float winkel = t+r;
    if(winkel>200){winkel -= 200; return winkel;}
    else{winkel += 200; return winkel;}
}

File: Python/grundlagen/winkel.py
from math import *

rho_deg = 180.0 / pi
rho_gon = 200.0 / pi

def rad2gon(p_rad):
    return p_rad * rho_gon

def rad2deg(p_rad):
    return p_rad * rho_deg

def gon2rad(p_gon):
    return p_gon / rho_gon

def gon2deg(p_gon):
    return rad2deg(gon2rad(p_gon))

def deg2rad(p_deg):
    return p_deg / rho_deg

def deg2gon(p_deg):
    return rad2gon(deg2rad(p_deg))
def cot(p_ctg):
    return cos(p_ctg)/sin(p_ctg)

def richtungswinkel_aus_richtung(t, r):
    t = t
    r = r
    winkel = t+r
    if winkel>200:
        winkel -= 200
        return winkel
    else:
        winkel += 200
        return winkel